src/common/pa_converters.c

```diff
@@ -935,13 +935,21 @@ static void Int32_To_Int24_Dither(
     void *sourceBuffer, signed int sourceStride,
     unsigned int count, struct PaUtilTriangularDitherGenerator *ditherGenerator )
 {
-    (void) destinationBuffer; /* unused parameters */
-    (void) destinationStride; /* unused parameters */
-    (void) sourceBuffer; /* unused parameters */
-    (void) sourceStride; /* unused parameters */
-    (void) count; /* unused parameters */
     (void) ditherGenerator; /* unused parameters */
-    /* IMPLEMENT ME */
+
+    PaInt32 *src = (PaInt32*)sourceBuffer;
+    signed char *dest = (signed char*)destinationBuffer;
+    PaInt32 dither;
+
+    while ( count-- )
+    {
+        /* REVIEW */
+        dither = PaUtil_Generate16BitTriangularDither(ditherGenerator);
+        *dest = (signed char) ((((*src) >> 1) + dither) >> 7);
```

Comment:
- review dither scaling
- assign to the three bytes of dest: `dest[0]`, `dest[1]`, `dest[2]`
- will need separate big/little endian cases